satpy/readers/gerb_l2_hr_h5.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2023
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+
+
+"""GERB L2 HR HDF5 reader.
+
+A reader for the Top of Atmosphere outgoing fluxes from the Geostationary Earth Radiation
+Budget instrument aboard the Meteosat Second Generation satellites.
+"""
+
+
+import logging
+from datetime import timedelta
+
+import dask.array as da
+import h5py
+import numpy as np
+import xarray as xr
+
+from satpy.readers.file_handlers import BaseFileHandler
+from satpy.resample import get_area_def
+
+LOG = logging.getLogger(__name__)
+
+
+def gerb_get_dataset(hfile, name, ds_info):
+    """
+    Load a GERB dataset in memory from a HDF5 file.
+
+    The routine takes into account the quantisation factor and fill values.
+    """
+    ds = hfile[name][...]
+    ds_attrs = hfile[name].attrs
+    ds_fill = ds_info['fill_value']
+    fill_mask = ds == ds_fill
+    if 'Quantisation Factor' in ds_attrs and 'Unit' in ds_attrs:
+        ds = ds*ds_attrs['Quantisation Factor']
+    else:
+        ds = ds*1.
+    ds[fill_mask] = np.nan
```

Comment:
This works for me. In the future when this is more dask based (maybe by using the helper HDF5 file handler) I think dask would like a `data_arr = data_arr.where(fill_mask)` but maybe we worry about that later. I think in the long run what I'm saying is that this reader should do as much in dask-land as possible. I'm not sure we have a hard requirement about when and where dask is used, but typically we do it as much as possible otherwise performance is generally wasted.

Reply:
Done.